db/migrations/1138.add_mivideo_tables.py

```diff
@@ -0,0 +1,14 @@
+from yoyo import step
+
+steps = [
+    step('''
+        CREATE TABLE IF NOT EXISTS mivideo_media_started_hourly (
+            id BIGINT AUTO_INCREMENT NOT NULL UNIQUE,
+            event_hour_utc VARCHAR(20) NOT NULL,
+            course_id BIGINT NOT NULL,
```

Comment:
I'm assuming this is Unizin ID? We may want to re-produce the Canvas course ID for simpler joining.

Reply:
When you say "Unizin ID", I guess you mean the ID from UDW, with the `17700000000000000` base ID.  This is not that.  It doesn't use the base ID.

`course_id` contains the Canvas course ID, as shown in the Caliper events' `$.object.extensions.kaf:course_id` properties.  These are the same course IDs that appear in URLs used to access the course in Canvas.  That is, in the URL `https://umich.instructure.com/courses/332500`, it is `332500`.

I think we definitely will want to join or make foreign key relationships with `course` and this table.  If the `course` table uses the longer IDs, then I guess we'll need to adjust.  I find the adding/subtracting of the base ID to be annoying.  I wish it were the same thing everywhere.

Would it be OK to add another column with the longer ID?

Reply:
We're no using the `canvas_id` as primary keys for the Canvas data tables, so you shouldn't need to do anything related to this if that's what you're getting back. I agree the relationship should be defined formally at some point, but we need to more thinking about linking tables that accumulate data and others that are always completely dropped.